guides/openstack.md

```diff
@@ -92,6 +93,220 @@ There are some differences in terminology between jclouds and OpenStack that sho
   </div>
 </div>
 
+## <a id="keystone"></a>Keystone v2-v3 authentication
+
+Openstack Keystone (aka: [OpenStack Identity Service](https://docs.openstack.org/keystone/latest/)) has major changes between v2 and v3 (detail. [Identity API v2.0 and v3 History](https://docs.openstack.org/keystone/latest/contributor/http-api.html)).
+
+Basically to login, you should provide:
+
+* On v2: *tenant*, *user*, *password*.
+* On v3: a *project* (new name for *tenant*), an authentication *domain* for this *project*, a *user*, an authentication *domain* for this *user* (the two domains can be different).
+
+jclouds provides backward compatibility between keystone v2-v3 ... but you should have following section in mind to fully understand the authentication on your Openstack platform (in addition of blog: [OpenStack Keystone V3 Support](https://jclouds.apache.org/blog/2018/01/16/keystone-v3/)).
+
+### v2
+
+This snippet:
+{% highlight java %}
+final Properties overrides = new Properties();
+overrides.put(KeystoneProperties.KEYSTONE_VERSION, "2");
+
+ContextBuilder.newBuilder("openstack-nova")
+   .endpoint("https://host:5000/v2.0")
+   .credentials("myTenant:foo", "bar")
+   .overrides(overrides)
+   .buildApi(NovaApi.class);
+{% endhighlight %}
+
+or 
+
+{% highlight java %}
+final Properties overrides = new Properties();
+overrides.put(KeystoneProperties.KEYSTONE_VERSION, "2");
+overrides.put(KeystoneProperties.TENANT_NAME, "myTenant");
+
+ContextBuilder.newBuilder("openstack-nova")
+   .endpoint("https://host:5000/v2.0")
+   .credentials("foo", "bar")
+   .overrides(overrides)
+   .buildApi(NovaApi.class);
+{% endhighlight %}
+
+Will produce when authentication needed: 
+
+    POST https://host:5000/v2.0/tokens HTTP/1.1
+    {
```

Comment:
Is there any particular benefit to trying to list the request so exactly here? In my experience, these are the parts of documentation that go stale very quickly.

Can we instead either give the code samples to the user and explain how they can configure logging to see what exactly is being sent (if that is necessary), or describe this a bit more abstractly, e.g. in form of a table?

Reply:
Configure logging to see what exactly is sent is simple (jclouds write the property to set to `true` and SLF4J in debug).

The (little) difficulty in jclouds authentication usage is the option to configure. It depends mainly of OpenStack platform configuration (v2, v3, domains, ...) ... and some manual REST tries on token endpoint could be required to see what is supported by the platform.

So these snippet are to help the user on jclouds usage when it has found its way of auhentication on OpenStack.